keanu-project/src/main/java/io/improbable/keanu/vertices/dbl/nonprobabilistic/diff/DualNumber.java

```diff
@@ -350,4 +351,84 @@ private DoubleTensor pluckFromPartial(DoubleTensor from, int... indices) {
             .reshape(pluckShape);
     }
 
+    public DualNumber ifThenElse(BooleanTensor predicate, DualNumber els) {
```

Comment:
A static method that takes if, then and else reads better I think.
```    
public static DualNumber ifThenElse(BooleanTensor predicate, DualNumber thn, DualNumber els){
        if (predicate.allTrue()) {
            return new DualNumber(thn.value, thn.getPartialDerivatives());
        } else if (predicate.allFalse()) {
            return new DualNumber(els.value, els.getPartialDerivatives());
        } else {
            PartialDerivatives mixedPartials = PartialDerivatives.ifThenElse(predicate, thn.getPartialDerivatives(), els.getPartialDerivatives());
            return new DualNumber(predicate.setDoubleIf(thn.value, els.value), mixedPartials);
        }
    }
```

Reply:
Cool, makes more sense too.